### PR TITLE
fix: fix 0x00 method id return to null

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
issue link: #1277 
## Check List
- transaction object with method id
#### Test
- unit test
#### Task
 - none

### CODE REVIEW

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex [hunk 1]
```diff
@@ -31,6 +32,14 @@ defmodule GodwokenExplorer.Graphql.Resolvers.Transaction do
     end
   end
 
+  def method_id(%{method_id: method_id}, _args, _resolution) do
+    case method_id |> to_string() do
+      "" -> {:ok, nil}
+      "0x00" -> {:ok, nil}
```
Comment (Keith-CY) [inline lib/godwoken_explorer/graphql/resolovers/transaction.ex:38]: It looks like a monkey patch? When will `method_id` be `0x00` in the workflow?

**Comment by Naupio** [comment]
this pr use "0x00" to replace the nil value of storage https://github.com/Magickbase/godwoken_explorer/pull/1063/files#diff-eb6de983469d4ff2615e61cdbe8078ca49515972c668ccf747dd616ed0e929d0R49

**Comment by Naupio** [comment]
and here use `method id is nil` condition to check the records which need to update method_id and method_name
https://github.com/Magickbase/godwoken_explorer/pull/1063/files#diff-eb6de983469d4ff2615e61cdbe8078ca49515972c668ccf747dd616ed0e929d0R76

**Comment by Keith-CY** [comment]
> this pr use "0x00" to replace the nil value of storage [#1063 (files)](https://github.com/Magickbase/godwoken_explorer/pull/1063/files#diff-eb6de983469d4ff2615e61cdbe8078ca49515972c668ccf747dd616ed0e929d0R49)

That was my negligence, why is `0x00` set as the default value of an invalid method id.
And as I remember, the `method id` could be `nil` for a long time, so default text `Send` was added in front-end https://github.com/Magickbase/godwoken_explorer/issues/999

**Comment by Keith-CY** [comment]
> and here use `method id is nil` condition to check the records which need to update method_id and method_name [#1063 (files)](https://github.com/Magickbase/godwoken_explorer/pull/1063/files#diff-eb6de983469d4ff2615e61cdbe8078ca49515972c668ccf747dd616ed0e929d0R76)

So resolving `0x00` method id to `nil` won't change the behavior of query here, am I right?

**Comment by Naupio** [comment]
> > and here use `method id is nil` condition to check the records which need to update method_id and method_name [#1063 (files)](https://github.com/Magickbase/godwoken_explorer/pull/1063/files#diff-eb6de983469d4ff2615e61cdbe8078ca49515972c668ccf747dd616ed0e929d0R76)
> 
> So resolving `0x00` method id to `nil` won't change the behavior of query here, am I right?
yes

**Comment by Keith-CY** [comment]
> > > and here use `method id is nil` condition to check the records which need to update method_id and method_name [#1063 (files)](https://github.com/Magickbase/godwoken_explorer/pull/1063/files#diff-eb6de983469d4ff2615e61cdbe8078ca49515972c668ccf747dd616ed0e929d0R76)
> > 
> > 
> > So resolving `0x00` method id to `nil` won't change the behavior of query here, am I right?
> > yes

Is it expected to filter `0x00` method id out when the search condition is `method id is nil`

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex (hunk 1 above)
Comment (zmcNotafraid) [reply lib/godwoken_explorer/graphql/resolovers/transaction.ex:38]: When the transaction is native transfer, the method_id should be nil not `0x00`

**Comment by Naupio** [comment]
change fix strategy:
fix `CheckUpdateTransactionMethodIdName` worker not use "0x00" placeholder for checking condition

File: lib/godwoken_explorer/graphql/types/transaction.ex [hunk 2]
```diff
@@ -395,41 +395,43 @@ defmodule GodwokenExplorer.Graphql.Types.Transaction do
   end
 
   object :transaction do
-    field :hash, :hash_full, description: "Godwoken transaction hash."
-    field :args, :chain_data, description: ""
+    field(:hash, :hash_full, description: "Godwoken transaction hash.")
```
Comment (Keith-CY) [inline lib/godwoken_explorer/graphql/types/transaction.ex:398]: Format updates should be removed in this PR, and open a new one if necessary

File: lib/godwoken_indexer/worker/check_update_transaction_method_id_name.ex [hunk 3]
```diff
@@ -82,6 +83,7 @@ defmodule GodwokenIndexer.Worker.CheckUpdateTransactionMethodIdName do
       where: a.type == :polyjuice_contract,
       inner_join: p in Polyjuice,
       on: t.hash == p.tx_hash,
+      where: fragment("length(?)", p.input) > 0,
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/worker/check_update_transaction_method_id_name.ex:86]: 直接input_size != 0 即可
```suggestion
      where: p.input_size != 0,
```

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

**Review by zmcNotafraid — APPROVED** [review]

**Comment by FrederLu** [comment]
It is found that the `Method` of this data is displayed as `0x00`, is it because the historical data has not been processed?
https://godwoken-explorer-ui-development.vercel.app/zh-CN/tx/0x63030ba1887aa4acea5f77e82e0f229fef9c5f9159afed0354d61e648a618b21
image

**Comment by Naupio** [comment]
> 

fixed, and pls check newest transaction

**Comment by FrederLu** [comment]
These two pieces of data also need to be reconfirmed.
https://godwoken-explorer-ui-development.vercel.app/tx/0xfd3db1e8d820187dc5f6b29686b0e68ead2340728537b37d17c3d247960bf179?tab=erc20Records
image

https://godwoken-explorer-ui-development.vercel.app/tx/0xdd604b3e71c074eb914fef08f485c409e3b1e29bcb853bfafabc1f1acf61d23c?search=0xdd604b3e71c074eb914fef08f485c409e3b1e29bcb853bfafabc1f1acf61d23c
image

**Comment by Naupio** [comment]
seems new transaction not fixed, add new commit to fixed here https://github.com/Magickbase/godwoken_explorer/pull/1298/commits/f8e723ed017601044c1a6bf71987cc61b8f6049c

> These two pieces of data also need to be reconfirmed. https://godwoken-explorer-ui-development.vercel.app/tx/0xfd3db1e8d820187dc5f6b29686b0e68ead2340728537b37d17c3d247960bf179?tab=erc20Records image
> 
> https://godwoken-explorer-ui-development.vercel.app/tx/0xdd604b3e71c074eb914fef08f485c409e3b1e29bcb853bfafabc1f1acf61d23c?search=0xdd604b3e71c074eb914fef08f485c409e3b1e29bcb853bfafabc1f1acf61d23c image

**Review by FrederLu — APPROVED** [review]

**Pull request merged**